src/UI/WB.UI.Designer/Controllers/QuestionnaireController.cs

```diff
@@ -17,20 +25,21 @@
 using WB.Core.BoundedContexts.Designer.Views.Questionnaire.Edit;
 using WB.Core.BoundedContexts.Designer.Views.Questionnaire.Edit.QuestionnaireInfo;
 using WB.Core.GenericSubdomains.Portable;
-using WB.Core.GenericSubdomains.Portable.Services;
 using WB.Core.Infrastructure.CommandBus;
 using WB.Core.Infrastructure.FileSystem;
-using WB.Core.SharedKernels.Questionnaire.Translations;
 using WB.UI.Designer.Code;
-using WB.UI.Designer.Code.ImportExport;
+using WB.UI.Designer.Controllers.Api.Designer;
 using WB.UI.Designer.Extensions;
 using WB.UI.Designer.Filters;
+using WB.UI.Designer.Models;
 using WB.UI.Designer.Resources;
+using WB.UI.Shared.Web.Services;
 
 namespace WB.UI.Designer.Controllers
 {
-    [Authorize]
+    [AuthorizeOrAnonymousQuestionnaire]

```

Comment:
Would you mind to return back Authorize attribute and mark specifically a few methods that should be visible with AuthorizeOrAnonymousQuestionnaire? Majority of methods in this file are marked with overriding Authorized and this class is partial that adds extra danger.

Reply:
I try this,  but if class has Authorize attribute, this attribute will be processed before attribute on method.

Reply:
And I split questionnaire list functionality to different class, to  save Authorize  attribute on operations with list

Reply:
Discussed